packages/react-native-web/src/exports/Modal/__tests__/index.js

```diff
@@ -426,4 +426,74 @@ describe('components/Modal', () => {
     expect(spy).toHaveBeenNthCalledWith(1, 'ref');
     expect(spy).toHaveBeenNthCalledWith(2, 'mount');
   });
+
+  test('escape key fires onRequestClose', () => {
+    const spy = jest.fn();
+
+    render(<Modal onRequestClose={spy} visible={true} />);
+
+    fireEvent.keyUp(document, { key: 'Escape' });
+
+    expect(spy).toHaveBeenCalledTimes(1);
+  });
+
+  test('escape key fires onRequestClose for top modal only', () => {
+    const spyA = jest.fn();
+    const spyB = jest.fn();
+
+    render(
+      <>
+        <Modal onRequestClose={spyA} visible={true} />
+        <Modal onRequestClose={spyB} visible={true} />
+      </>
+    );
+
+    fireEvent.keyUp(document, { key: 'Escape' });
+
+    expect(spyA).toHaveBeenCalledTimes(0);
+    expect(spyB).toHaveBeenCalledTimes(1);
+  });
+
+  test('escape key fires onRequestClose for top modal only with animation', () => {
```

Comment:
I think this test also needs to pass if animations are not used

Reply:
If an animation is used, we wait until the animation completes to send the `onDismiss` event from the modal's animation layer.  This means if an animation is used it will be considered "active" until the animation completes.  While a modal is active it will receive the "escape" events.  This is to match up with how ReactNative & iOS seem to do this - at least that's what I vaguely recall.  This could be incorrect or an undesired behavior, I'm not sure.

If we don't want that behavior and instead want a modal to be inactive as soon as it's requested but before it actually closes we can do that, but it'd be a slightly larger change.

Is that what you're requesting?